cmd/kube-apiserver/app/server.go

```diff
@@ -527,7 +528,7 @@ func buildGenericConfig(
 }
 
 // BuildAuthenticator constructs the authenticator
-func BuildAuthenticator(s *options.ServerRunOptions, extclient clientgoclientset.Interface, versionedInformer clientgoinformers.SharedInformerFactory) (authenticator.Request, *spec.SecurityDefinitions, error) {
+func BuildAuthenticator(s *options.ServerRunOptions, EgressSelector *egressselector.EgressSelector, extclient clientgoclientset.Interface, versionedInformer clientgoinformers.SharedInformerFactory) (authenticator.Request, *spec.SecurityDefinitions, error) {
```

Comment:
* Consider moving the EgressSelector into ServerRunOptions
* Consider having a no-op implementation (so no `if nil` checks needed)

Reply:
> * Consider moving the EgressSelector into ServerRunOptions
ack
> * Consider having a no-op implementation (so no `if nil` checks needed)
See comment below. I'm not sure if that's possible since we're replacing the Dialer instead of proxying through with the Dialer.

Reply:
Maybe the function signature could take the existing dialer, so that the no-op implementation can just return it unmodified? That'd also permit you to do the thing in my other comment if it ever becomes necessary.

Reply:
@lavalamp: After digging into this a bit more, I'm not sure if it'd be a good idea to move this into ServerRunOptions. The objects in that struct are heavily coupled to apiserver's startup flags and store startup configs read from the CLI. EgressSelector is a class containing more complex functions that I feel would be more suitable passed around as a stand alone class.

Reply:
As discussed offline, let's move the no-op egress selector implementation to a separate PR since it will require a refactor of all egress endpoints.